src/typeDefinition.ts

```diff
@@ -71,6 +71,9 @@ export class TypeDefinition {
             base.allOf.forEach((p) => {
                 if (p.$ref) {
                     p = this.searchRef(process, p.$ref).targetSchema;
+                    if (p.allOf && p.allOf.some((allOfChild: JsonSchemaOrg.Schema) => !!allOfChild.$ref)) {
```

Comment:
I think this line is unnecessary.
The `checkSchema` method check the near condition.

Reply:
You're right. This additional check can be removed.